File: crossecore-csharp/Java/Java/Java_/InterfaceDeclarationBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class InterfaceDeclarationBase 
	:TypeDeclarationImpl, InterfaceDeclaration
	{
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.INTERFACEDECLARATION;
		}
		
		
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTS:
					return comments;
				case Java_PackageImpl.INTERFACEDECLARATION_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.INTERFACEDECLARATION_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.INTERFACEDECLARATION_NAME:
					return name;
				case Java_PackageImpl.INTERFACEDECLARATION_PROXY:
					return proxy;
				case Java_PackageImpl.INTERFACEDECLARATION_USAGESINIMPORTS:
					return usagesInImports;
				case Java_PackageImpl.INTERFACEDECLARATION_ABSTRACTTYPEDECLARATION:
					return abstractTypeDeclaration;
				case Java_PackageImpl.INTERFACEDECLARATION_ANNOTATIONS:
					return annotations;
				case Java_PackageImpl.INTERFACEDECLARATION_ANONYMOUSCLASSDECLARATIONOWNER:
					return anonymousClassDeclarationOwner;
				case Java_PackageImpl.INTERFACEDECLARATION_MODIFIER:
					return modifier;
				case Java_PackageImpl.INTERFACEDECLARATION_USAGESINTYPEACCESS:
					return usagesInTypeAccess;
				case Java_PackageImpl.INTERFACEDECLARATION_BODYDECLARATIONS:
					return bodyDeclarations;
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTSBEFOREBODY:
					return commentsBeforeBody;
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTSAFTERBODY:
					return commentsAfterBody;
				case Java_PackageImpl.INTERFACEDECLARATION_PACKAGE:
					return package;
				case Java_PackageImpl.INTERFACEDECLARATION_SUPERINTERFACES:
					return superInterfaces;
				case Java_PackageImpl.INTERFACEDECLARATION_TYPEPARAMETERS:
					return typeParameters;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_NAME:
					name = (string) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_PROXY:
					proxy = (bool) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_USAGESINIMPORTS:
					usagesInImports.Clear();
					usagesInImports.AddRange(((List<EObject>)newValue)?.Cast<ImportDeclaration>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_ABSTRACTTYPEDECLARATION:
					abstractTypeDeclaration = (AbstractTypeDeclaration) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_ANNOTATIONS:
					annotations.Clear();
					annotations.AddRange(((List<EObject>)newValue)?.Cast<Annotation>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_ANONYMOUSCLASSDECLARATIONOWNER:
					anonymousClassDeclarationOwner = (AnonymousClassDeclaration) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_MODIFIER:
					modifier = (Modifier) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_USAGESINTYPEACCESS:
					usagesInTypeAccess.Clear();
					usagesInTypeAccess.AddRange(((List<EObject>)newValue)?.Cast<TypeAccess>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_BODYDECLARATIONS:
					bodyDeclarations.Clear();
					bodyDeclarations.AddRange(((List<EObject>)newValue)?.Cast<BodyDeclaration>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTSBEFOREBODY:
					commentsBeforeBody.Clear();
					commentsBeforeBody.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_COMMENTSAFTERBODY:
					commentsAfterBody.Clear();
					commentsAfterBody.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_PACKAGE:
					package = (Package) newValue;
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_SUPERINTERFACES:
					superInterfaces.Clear();
					superInterfaces.AddRange(((List<EObject>)newValue)?.Cast<TypeAccess>());
					return;
				case Java_PackageImpl.INTERFACEDECLARATION_TYPEPARAMETERS:
					typeParameters.Clear();
					typeParameters.AddRange(((List<EObject>)newValue)?.Cast<TypeParameter>());
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/InterfaceDeclarationBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {InterfaceDeclaration} from "./InterfaceDeclaration";
import {TypeDeclarationImpl} from "./TypeDeclarationImpl";
		
			export class InterfaceDeclarationBase
			extends TypeDeclarationImpl
			implements InterfaceDeclaration
			{

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return InterfaceDeclarationBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.INTERFACEDECLARATION_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.INTERFACEDECLARATION_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.INTERFACEDECLARATION_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.INTERFACEDECLARATION_NAME:
							return this.name;
						case Java_PackageLiterals.INTERFACEDECLARATION_PROXY:
							return this.proxy;
						case Java_PackageLiterals.INTERFACEDECLARATION_USAGESINIMPORTS:
							return this.usagesInImports;
						case Java_PackageLiterals.INTERFACEDECLARATION_ABSTRACTTYPEDECLARATION:
							return this.abstractTypeDeclaration;
						case Java_PackageLiterals.INTERFACEDECLARATION_ANNOTATIONS:
							return this.annotations;
						case Java_PackageLiterals.INTERFACEDECLARATION_ANONYMOUSCLASSDECLARATIONOWNER:
							return this.anonymousClassDeclarationOwner;
						case Java_PackageLiterals.INTERFACEDECLARATION_MODIFIER:
							return this.modifier;
						case Java_PackageLiterals.INTERFACEDECLARATION_USAGESINTYPEACCESS:
							return this.usagesInTypeAccess;
						case Java_PackageLiterals.INTERFACEDECLARATION_BODYDECLARATIONS:
							return this.bodyDeclarations;
						case Java_PackageLiterals.INTERFACEDECLARATION_COMMENTSBEFOREBODY:
							return this.commentsBeforeBody;
						case Java_PackageLiterals.INTERFACEDECLARATION_COMMENTSAFTERBODY:
							return this.commentsAfterBody;
						case Java_PackageLiterals.INTERFACEDECLARATION_PACKAGE:
							return this.package;
						case Java_PackageLiterals.INTERFACEDECLARATION_SUPERINTERFACES:
							return this.superInterfaces;
						case Java_PackageLiterals.INTERFACEDECLARATION_TYPEPARAMETERS:
							return this.typeParameters;
					}
					//return this.eGetFromTypeDeclaration(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
